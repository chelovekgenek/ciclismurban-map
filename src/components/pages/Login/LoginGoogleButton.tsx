import React, { useState, useCallback } from "react"
import { GoogleLogin, GoogleLoginResponse, GoogleLoginProps } from "react-google-login"

import { IDispatchProps } from "./LoginGoogleButton.container"
import * as LoginStyles from "./Login.styled"

interface IProps extends IDispatchProps {}

export const LoginGoogleButton: React.FC<IProps> = ({ login }) => {
  const [disabled, setDisabled] = useState(false)
  const handleSuccess = useCallback<GoogleLoginProps["onSuccess"]>(
    r => (r as GoogleLoginResponse).tokenObj && login((r as GoogleLoginResponse).tokenObj.id_token),
    [login],
  )
  const handleFailure = useCallback(() => setDisabled(true), [])
  return (
    <GoogleLogin
      clientId={process.env.REACT_APP_GOOGLE_OAUTH_KEY as string}
      render={renderProps => (
        <LoginStyles.Button icon="google" onClick={renderProps.onClick} disabled={renderProps.disabled || disabled}>
          Google
        </LoginStyles.Button>
      )}
      onSuccess={handleSuccess}
      onFailure={handleFailure}
      cookiePolicy="single_host_origin"
    />
  )
}
