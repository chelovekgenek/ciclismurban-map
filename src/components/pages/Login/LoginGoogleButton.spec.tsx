import React from "react"
import { shallow } from "enzyme"

import { LoginGoogleButton } from "./LoginGoogleButton"

describe("LoginFacebookButton", () => {
  const render = (props: Partial<React.ComponentProps<typeof LoginGoogleButton>> = {}) =>
    shallow(<LoginGoogleButton login={jest.fn() as any} {...props} />)
  it("should match snapshot", () => {
    expect(render()).toMatchSnapshot()
  })
})
