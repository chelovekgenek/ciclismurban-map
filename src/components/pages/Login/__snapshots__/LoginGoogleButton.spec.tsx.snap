// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFacebookButton should match snapshot 1`] = `
<e
  accessType="online"
  buttonText="Sign in with Google"
  clientId="900137834161-n0kou6bi45thoqauun42u6m1fn7k21gi.apps.googleusercontent.com"
  cookiePolicy="single_host_origin"
  disabledStyle={
    Object {
      "opacity": 0.6,
    }
  }
  fetchBasicProfile={true}
  icon={true}
  isSignedIn={false}
  jsSrc="https://apis.google.com/js/api.js"
  onFailure={[Function]}
  onRequest={[Function]}
  onSuccess={[Function]}
  prompt=""
  render={[Function]}
  scope="profile email"
  tag="button"
  theme="light"
  type="button"
  uxMode="popup"
/>
`;
